tests/periph_i2c/tests/03__periph_i2c_error_codes.robot

```diff
@@ -0,0 +1,26 @@
+*** Settings ***
+Documentation       Data driven tests to verify writing to a valid register.
+
+Suite Setup         Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+...                                 I2C Acquire
+Test Teardown       I2C Release
+
+Resource            periph_i2c.keywords.txt
+Resource            api_shell.keywords.txt
+
+Force Tags          periph  i2c
+
+*** Test Cases ***
+Wrong Address Should ENXIO
```

Comment:
I think the exact error code doesn't have value/meaning to the user/reader of this test, simpler and more descriptive (to me) sounds: `Read Register From Wrong Address Should Fail`

Sounds like you can also add a similar test like `Read Bytes From Wrong Address Should Fail`?!

Btw. `wrong` sound so lazy,  maybe `invalid` would fit better in this context?

Reply:
Yes to invalid but we should distiguish between error codes.  Our API states it for one, and there are practical reasons why one would need to know the difference between address NACK and data NACK.